code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -213,26 +214,40 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 /obj/item/clothing/mask/cigarette/process()
 	var/turf/location = get_turf(src)
 	var/mob/living/M = loc
-	if(isliving(loc))
-		M.IgniteMob()
-	smoketime--
-	if(smoketime < 1)
-		new type_butt(location)
-		if(ismob(loc))
-			M << "<span class='notice'>Your [name] goes out.</span>"
-			M.unEquip(src, 1)	//un-equip it so the overlays can update //Force the un-equip so the overlays update
-		qdel(src)
-		return
-	open_flame()
+	if(!e_cig)
+		if(isliving(loc))
+			M.IgniteMob()
+		smoketime--
+		if(smoketime < 1)
+			new type_butt(location)
+			if(ismob(loc))
+				M << "<span class='notice'>Your [name] goes out.</span>"
+				M.unEquip(src, 1)	//un-equip it so the overlays can update //Force the un-equip so the overlays update

```

Comment:
Hello, you've reached the Department of Redundancy Department. I'm Agent Mike Michael Michaelson, your Agent for today, how can I help you?